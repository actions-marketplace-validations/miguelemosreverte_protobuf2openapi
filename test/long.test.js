const path = require('path');
const convert = require('../lib/convert');

describe('long', () => {
  test('string by default', async () => {
    const protopath = path.resolve(__dirname, '__fixtures__/long.proto');
    const converted = await convert({
      files: [protopath],
    });
    expect(converted).toMatchSnapshot();
  });
  test('use option to change to integer', async () => {
    const protopath = path.resolve(__dirname, '__fixtures__/long.proto');
    const converted = await convert({
      files: [protopath],
      long: 'number',
    });
    expect(converted).toMatchSnapshot();
  });
});
