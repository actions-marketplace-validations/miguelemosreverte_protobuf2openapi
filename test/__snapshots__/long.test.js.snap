// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`long string by default 1`] = `
Object {
  "components": Object {
    "schemas": Object {
      "long.ABitOfEverything": Object {
        "properties": Object {
          "double_value": Object {
            "format": "double",
            "type": "string",
          },
          "fixed32_value": Object {
            "format": "int32",
            "type": "integer",
          },
          "fixed64_value": Object {
            "format": "int64",
            "type": "string",
          },
          "float_value": Object {
            "format": "float",
            "type": "number",
          },
          "int32_value": Object {
            "format": "int32",
            "type": "integer",
          },
          "int64_value": Object {
            "format": "int64",
            "type": "string",
          },
          "sfixed32_value": Object {
            "format": "int64",
            "type": "integer",
          },
          "sfixed64_value": Object {
            "format": "int64",
            "type": "string",
          },
          "sint32_value": Object {
            "format": "int32",
            "type": "integer",
          },
          "sint64_value": Object {
            "format": "int64",
            "type": "string",
          },
          "uint32_value": Object {
            "format": "int32",
            "type": "integer",
          },
          "uint64_value": Object {
            "format": "int64",
            "type": "string",
          },
        },
        "required": Array [],
        "type": "object",
      },
    },
  },
  "info": Object {
    "description": "",
    "title": "long",
    "version": "1",
  },
  "openapi": "3.0.0",
  "paths": Object {},
}
`;

exports[`long use option to change to integer 1`] = `
Object {
  "components": Object {
    "schemas": Object {
      "long.ABitOfEverything": Object {
        "properties": Object {
          "double_value": Object {
            "format": "double",
            "type": "number",
          },
          "fixed32_value": Object {
            "format": "int32",
            "type": "integer",
          },
          "fixed64_value": Object {
            "format": "int64",
            "type": "integer",
          },
          "float_value": Object {
            "format": "float",
            "type": "number",
          },
          "int32_value": Object {
            "format": "int32",
            "type": "integer",
          },
          "int64_value": Object {
            "format": "int64",
            "type": "integer",
          },
          "sfixed32_value": Object {
            "format": "int64",
            "type": "integer",
          },
          "sfixed64_value": Object {
            "format": "int64",
            "type": "integer",
          },
          "sint32_value": Object {
            "format": "int32",
            "type": "integer",
          },
          "sint64_value": Object {
            "format": "int64",
            "type": "integer",
          },
          "uint32_value": Object {
            "format": "int32",
            "type": "integer",
          },
          "uint64_value": Object {
            "format": "int64",
            "type": "integer",
          },
        },
        "required": Array [],
        "type": "object",
      },
    },
  },
  "info": Object {
    "description": "",
    "title": "long",
    "version": "1",
  },
  "openapi": "3.0.0",
  "paths": Object {},
}
`;
